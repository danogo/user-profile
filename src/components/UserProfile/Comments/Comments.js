import React from 'react';
import styles from './Comments.module.css';
import picture from './pic.jpg';
import moment from 'moment';

// configure moment to use single letter date format
moment.updateLocale('en', {
    relativeTime: {
      future: 'in %s',
      past: '%s',
      s:  'a few seconds',
      ss: '%ss',
      m:  'a minute',
      mm: '%dm',
      h:  'an hour',
      hh: '%dh',
      d:  'a day',
      dd: '%dd',
      M:  'a month',
      MM: '%dM',
      y:  'a year',
      yy: '%dY'
    }
  });

const Comments = (props) => {
  const { onInput, onToggleComments, comments, showComments } = props;

  const renderComments = (commentsArray) => {
    return commentsArray.map( (el, i) => {
      return (
        <li key={i}>
          <div className={styles.authorBox}>
            <img src={picture} alt='author of the comment'/>
            <div className={styles.author}>
              <span className={styles.name}>{el.username}</span>
              <span className={styles.date}>{moment(parseInt(el.date, 10)).fromNow()}</span>
            </div>
          </div>
          <p className={styles.content}>{el.content}</p>
        </li>
      )
    });
  }

  return (
    <div className={styles.Comments}>
      <button onClick={onToggleComments} className={showComments ? styles.btnComments : `${styles.btnCommentsHidden} ${styles.btnComments}`}>
        {showComments ? 'Hide comments' : 'Show comments'} 
        <span className={styles.numOfCom}>{`(${comments.length})`}</span>
      </button>
      <ul id="comList" className={showComments ? '' : styles.hidden} tabIndex='0'>
        {renderComments(comments)} 
      </ul>
      <div className={styles.inputBox}>
        <input onKeyPress={onInput} type="text" className={styles.inputComment} placeholder='Add a comment'/>
      </div>
    </div>
  ); 
};

export default Comments;

