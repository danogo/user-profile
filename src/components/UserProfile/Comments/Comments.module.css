.Comments {
  width: 93%;
  background-color: #FFFFFF;
  margin: 0 auto;
  border-radius: .5rem;
  box-shadow: 0 0 4px #ACACAC50;
  padding-top: 2.5rem;
  position: relative;
  top: 3rem;
  font-family: 'Montserrat';
  padding-right: .6rem;
}

.Comments ul {
  list-style: none;
  overflow-y: scroll;
  height: 33.1rem;
  padding-right: .8rem;
  transition: height .4s ease-in-out;
}

.Comments ul::-webkit-scrollbar {
  width: 5px;
  background-color: #D8D8D860;
  border-radius: 1rem;
}

.Comments ul::-webkit-scrollbar-thumb {
  background-color: #002C7130;
  border-radius: 1rem;
}

.Comments ul:focus,
.Comments ul:active {
  outline: 2px solid #FFA640;
  color: #FFBC70;
}

.Comments li {
  border-top: 1px solid #EBEBEB;
  padding: 3rem 2rem 2.55rem 2rem;
}

.Comments li:first-of-type {
  border-top: none;
  padding-top: 0;
}

.btnComments {
  border: none;
  background: none;
  text-decoration: underline;
  font-size: 1.4rem;
  color: #FFA640;
  letter-spacing: 0;
  font-family: 'Montserrat';
  cursor: pointer;
  margin-bottom: 3rem;
  margin-left: 2rem;
}

.btnComments:hover {
  color: #FFBC70;
}

.btnComments:focus,
.btnComments:active {
  outline: 2px solid #FFA640;
  outline-offset: 2px;
  color: #FFBC70;
}

.Comments img {
  width: 4rem;
  height: 4rem;
  border-radius: 100%;
  margin-right: 1.1rem;
}

.authorBox {
  display: flex;
}

.author {
 flex-grow: 1;
 display: flex;
 align-items: center;
 justify-content: space-between;
}

.content {
  padding-left: 5.1rem;
  padding-right: 2.7rem;
  font-size: 14px;
  color: #444444;
  letter-spacing: 0;
  line-height: 21px;
  overflow-wrap: break-word;
  word-wrap: break-word;
  hyphens: auto;
}

.name {
  font-size: 1.4rem;
  color: #002C71;
  letter-spacing: 0;
  font-weight: bold;
}

.date {
  opacity: 0.5;
  font-size: 1.4rem;
  color: #002C71;
  letter-spacing: 0;
  font-weight: bold;
}

.inputBox {
  padding: 4rem 2.5rem 3.5rem 2.5rem;
  position: absolute;
  width: 100%;
  top: 100%;
  left: 0;
  box-shadow: 0 0 4px #ACACAC50;
  border-radius: 5px;
}

.inputComment {
  border: 0;
  border-bottom: 2px solid #DBE0E8;
  width: 100%;
  color: #8298B9;
  padding: .5rem 0;
  font-size: 1.8rem;
  letter-spacing: 0;
  font-family: 'Montserrat';
  transition: border-bottom-color .3s ease-in;
}

.inputComment::placeholder {
  font-size: 1.8rem;
  color: #8298B9;
  font-family: 'Montserrat';
}

.inputComment:focus,
.inputComment:active {
  border-bottom-color: #FFA640;
  outline: none;
}

.CommentsHidden ul {
  /* display: none; */
  height: 0;
  transition: height .4s ease-in-out;
  outline: none;
}

.CommentsHidden ul li:first-of-type {
  border-top: none;
}

@media only screen and (max-width: 550px) {
  .btnComments {
    display: block;
    margin-left: auto;
    margin-right: auto;
    padding-left: 0;
  }
}