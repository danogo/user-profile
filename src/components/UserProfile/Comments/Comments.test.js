import React from 'react';
import { shallow } from 'enzyme';
import Comments from './Comments';

describe('Comments', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      onInput: jest.fn(),
      onToggleComments: jest.fn(),
      comments: [
        {
          "username": "Mike Ross",
          "date": "1539088547946",
          "content": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Architecto autem nam quae delectus maxime inventore odio libero provident obcaecati a."
        },
        {
          "username": "Bob Ross",
          "date": "1529588547946",
          "content": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Architecto autem nam quae delectus maxime inventore odio libero provident obcaecati a."
        },
      ],
      showComments: true
    }
    wrapper = shallow(<Comments {...mockProps}/>);
  });

  test('component renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('renders div container', () => {
    const divs = wrapper.find('div');
    expect(divs.length).toBeGreaterThan(0);
  });

  test('contains all children components', () => {
    const div = wrapper.find('div').first();
    expect(div.children()).toEqual(wrapper.children());
    expect(wrapper.children().length).toBe(3);
  });

  test('receives all onClick handlers', () => {
    expect(wrapper.find('#inputComment').props().onKeyPress).toBe(mockProps.onInput);
    expect(wrapper.find('.btnComments').props().onClick).toBe(mockProps.onToggleComments);
  });

  test('displays comments by default', () => {
    expect(wrapper.props().className).toBe('Comments');
    expect(wrapper.find('#comList').children().length).toBe(mockProps.comments.length)
  });

  test('comments are sorted properly', () => {
    const sortedComments = mockProps.comments.sort((elA, elB) => {
      return parseInt(elA.date, 10) - parseInt(elB.date, 10);
    });
    // console.log(wrapper.find('#comList').childAt(0).find('.name').props())
    // console.log(sortedComments)
    expect(wrapper.find('#comList').childAt(0).find('.name').text()).toBe(sortedComments[0].username);
  });

  test('callback is called when user clicks button to hide comments', () => {
    const btnComments = wrapper.find('.btnComments');
    btnComments.prop('onClick')();
    expect(mockProps.onToggleComments).toHaveBeenCalledTimes(1);
  });

  describe('showComments prop is false', () => {
    beforeEach(() => {
      mockProps.showComments = false;
      wrapper = shallow(<Comments {...mockProps}/>);
    });
    test('comments are not visible', () => {
      expect(wrapper.props().className).toBe('CommentsHidden Comments');
    });
    test('button has "Show comments(numOfComments)" text  when hidden', () => {
      const btnToggleComments = wrapper.find('.btnComments');
      expect(btnToggleComments.text()).toEqual(`Show comments(${mockProps.comments.length})`);
    });
  });
});