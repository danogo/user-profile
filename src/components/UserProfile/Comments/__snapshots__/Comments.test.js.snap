// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Comments
    comments={
      Array [
        Object {
          "content": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Architecto autem nam quae delectus maxime inventore odio libero provident obcaecati a.",
          "date": "1529588547946",
          "username": "Bob Ross",
        },
        Object {
          "content": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Architecto autem nam quae delectus maxime inventore odio libero provident obcaecati a.",
          "date": "1539088547946",
          "username": "Mike Ross",
        },
      ]
    }
    onInput={[MockFunction]}
    onToggleComments={[MockFunction]}
    showComments={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="btnComments"
          onClick={[MockFunction]}
        >
          Hide comments
          <span
            className="numOfCom"
          >
            (2)
          </span>
        </button>,
        <ul
          id="comList"
          tabIndex="0"
        >
          <li>
            <div
              className="authorBox"
            >
              <img
                alt="author of the comment"
                src="pic.jpg"
              />
              <div
                className="author"
              >
                <span
                  className="name"
                >
                  Bob Ross
                </span>
                <span
                  className="date"
                >
                  4M
                </span>
              </div>
            </div>
            <p
              className="content"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Architecto autem nam quae delectus maxime inventore odio libero provident obcaecati a.
            </p>
          </li>
          <li>
            <div
              className="authorBox"
            >
              <img
                alt="author of the comment"
                src="pic.jpg"
              />
              <div
                className="author"
              >
                <span
                  className="name"
                >
                  Mike Ross
                </span>
                <span
                  className="date"
                >
                  13d
                </span>
              </div>
            </div>
            <p
              className="content"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Architecto autem nam quae delectus maxime inventore odio libero provident obcaecati a.
            </p>
          </li>
        </ul>,
        <div
          className="inputBox"
        >
          <input
            className="inputComment"
            id="inputComment"
            onKeyPress={[MockFunction]}
            placeholder="Add a comment"
            type="text"
          />
        </div>,
      ],
      "className": "Comments",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Hide comments",
            <span
              className="numOfCom"
            >
              (2)
            </span>,
          ],
          "className": "btnComments",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": Array [
          "Hide comments",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "(2)",
              "className": "numOfCom",
            },
            "ref": null,
            "rendered": "(2)",
            "type": "span",
          },
        ],
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <div
                className="authorBox"
              >
                <img
                  alt="author of the comment"
                  src="pic.jpg"
                />
                <div
                  className="author"
                >
                  <span
                    className="name"
                  >
                    Bob Ross
                  </span>
                  <span
                    className="date"
                  >
                    4M
                  </span>
                </div>
              </div>
              <p
                className="content"
              >
                Lorem ipsum dolor sit amet consectetur adipisicing elit. Architecto autem nam quae delectus maxime inventore odio libero provident obcaecati a.
              </p>
            </li>,
            <li>
              <div
                className="authorBox"
              >
                <img
                  alt="author of the comment"
                  src="pic.jpg"
                />
                <div
                  className="author"
                >
                  <span
                    className="name"
                  >
                    Mike Ross
                  </span>
                  <span
                    className="date"
                  >
                    13d
                  </span>
                </div>
              </div>
              <p
                className="content"
              >
                Lorem ipsum dolor sit amet consectetur adipisicing elit. Architecto autem nam quae delectus maxime inventore odio libero provident obcaecati a.
              </p>
            </li>,
          ],
          "id": "comList",
          "tabIndex": "0",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="authorBox"
                >
                  <img
                    alt="author of the comment"
                    src="pic.jpg"
                  />
                  <div
                    className="author"
                  >
                    <span
                      className="name"
                    >
                      Bob Ross
                    </span>
                    <span
                      className="date"
                    >
                      4M
                    </span>
                  </div>
                </div>,
                <p
                  className="content"
                >
                  Lorem ipsum dolor sit amet consectetur adipisicing elit. Architecto autem nam quae delectus maxime inventore odio libero provident obcaecati a.
                </p>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <img
                      alt="author of the comment"
                      src="pic.jpg"
                    />,
                    <div
                      className="author"
                    >
                      <span
                        className="name"
                      >
                        Bob Ross
                      </span>
                      <span
                        className="date"
                      >
                        4M
                      </span>
                    </div>,
                  ],
                  "className": "authorBox",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "author of the comment",
                      "src": "pic.jpg",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <span
                          className="name"
                        >
                          Bob Ross
                        </span>,
                        <span
                          className="date"
                        >
                          4M
                        </span>,
                      ],
                      "className": "author",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Bob Ross",
                          "className": "name",
                        },
                        "ref": null,
                        "rendered": "Bob Ross",
                        "type": "span",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "4M",
                          "className": "date",
                        },
                        "ref": null,
                        "rendered": "4M",
                        "type": "span",
                      },
                    ],
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Architecto autem nam quae delectus maxime inventore odio libero provident obcaecati a.",
                  "className": "content",
                },
                "ref": null,
                "rendered": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Architecto autem nam quae delectus maxime inventore odio libero provident obcaecati a.",
                "type": "p",
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="authorBox"
                >
                  <img
                    alt="author of the comment"
                    src="pic.jpg"
                  />
                  <div
                    className="author"
                  >
                    <span
                      className="name"
                    >
                      Mike Ross
                    </span>
                    <span
                      className="date"
                    >
                      13d
                    </span>
                  </div>
                </div>,
                <p
                  className="content"
                >
                  Lorem ipsum dolor sit amet consectetur adipisicing elit. Architecto autem nam quae delectus maxime inventore odio libero provident obcaecati a.
                </p>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <img
                      alt="author of the comment"
                      src="pic.jpg"
                    />,
                    <div
                      className="author"
                    >
                      <span
                        className="name"
                      >
                        Mike Ross
                      </span>
                      <span
                        className="date"
                      >
                        13d
                      </span>
                    </div>,
                  ],
                  "className": "authorBox",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "author of the comment",
                      "src": "pic.jpg",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <span
                          className="name"
                        >
                          Mike Ross
                        </span>,
                        <span
                          className="date"
                        >
                          13d
                        </span>,
                      ],
                      "className": "author",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Mike Ross",
                          "className": "name",
                        },
                        "ref": null,
                        "rendered": "Mike Ross",
                        "type": "span",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "13d",
                          "className": "date",
                        },
                        "ref": null,
                        "rendered": "13d",
                        "type": "span",
                      },
                    ],
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Architecto autem nam quae delectus maxime inventore odio libero provident obcaecati a.",
                  "className": "content",
                },
                "ref": null,
                "rendered": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Architecto autem nam quae delectus maxime inventore odio libero provident obcaecati a.",
                "type": "p",
              },
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="inputComment"
            id="inputComment"
            onKeyPress={[MockFunction]}
            placeholder="Add a comment"
            type="text"
          />,
          "className": "inputBox",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "inputComment",
            "id": "inputComment",
            "onKeyPress": [MockFunction],
            "placeholder": "Add a comment",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="btnComments"
            onClick={[MockFunction]}
          >
            Hide comments
            <span
              className="numOfCom"
            >
              (2)
            </span>
          </button>,
          <ul
            id="comList"
            tabIndex="0"
          >
            <li>
              <div
                className="authorBox"
              >
                <img
                  alt="author of the comment"
                  src="pic.jpg"
                />
                <div
                  className="author"
                >
                  <span
                    className="name"
                  >
                    Bob Ross
                  </span>
                  <span
                    className="date"
                  >
                    4M
                  </span>
                </div>
              </div>
              <p
                className="content"
              >
                Lorem ipsum dolor sit amet consectetur adipisicing elit. Architecto autem nam quae delectus maxime inventore odio libero provident obcaecati a.
              </p>
            </li>
            <li>
              <div
                className="authorBox"
              >
                <img
                  alt="author of the comment"
                  src="pic.jpg"
                />
                <div
                  className="author"
                >
                  <span
                    className="name"
                  >
                    Mike Ross
                  </span>
                  <span
                    className="date"
                  >
                    13d
                  </span>
                </div>
              </div>
              <p
                className="content"
              >
                Lorem ipsum dolor sit amet consectetur adipisicing elit. Architecto autem nam quae delectus maxime inventore odio libero provident obcaecati a.
              </p>
            </li>
          </ul>,
          <div
            className="inputBox"
          >
            <input
              className="inputComment"
              id="inputComment"
              onKeyPress={[MockFunction]}
              placeholder="Add a comment"
              type="text"
            />
          </div>,
        ],
        "className": "Comments",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Hide comments",
              <span
                className="numOfCom"
              >
                (2)
              </span>,
            ],
            "className": "btnComments",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": Array [
            "Hide comments",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "(2)",
                "className": "numOfCom",
              },
              "ref": null,
              "rendered": "(2)",
              "type": "span",
            },
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <div
                  className="authorBox"
                >
                  <img
                    alt="author of the comment"
                    src="pic.jpg"
                  />
                  <div
                    className="author"
                  >
                    <span
                      className="name"
                    >
                      Bob Ross
                    </span>
                    <span
                      className="date"
                    >
                      4M
                    </span>
                  </div>
                </div>
                <p
                  className="content"
                >
                  Lorem ipsum dolor sit amet consectetur adipisicing elit. Architecto autem nam quae delectus maxime inventore odio libero provident obcaecati a.
                </p>
              </li>,
              <li>
                <div
                  className="authorBox"
                >
                  <img
                    alt="author of the comment"
                    src="pic.jpg"
                  />
                  <div
                    className="author"
                  >
                    <span
                      className="name"
                    >
                      Mike Ross
                    </span>
                    <span
                      className="date"
                    >
                      13d
                    </span>
                  </div>
                </div>
                <p
                  className="content"
                >
                  Lorem ipsum dolor sit amet consectetur adipisicing elit. Architecto autem nam quae delectus maxime inventore odio libero provident obcaecati a.
                </p>
              </li>,
            ],
            "id": "comList",
            "tabIndex": "0",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="authorBox"
                  >
                    <img
                      alt="author of the comment"
                      src="pic.jpg"
                    />
                    <div
                      className="author"
                    >
                      <span
                        className="name"
                      >
                        Bob Ross
                      </span>
                      <span
                        className="date"
                      >
                        4M
                      </span>
                    </div>
                  </div>,
                  <p
                    className="content"
                  >
                    Lorem ipsum dolor sit amet consectetur adipisicing elit. Architecto autem nam quae delectus maxime inventore odio libero provident obcaecati a.
                  </p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <img
                        alt="author of the comment"
                        src="pic.jpg"
                      />,
                      <div
                        className="author"
                      >
                        <span
                          className="name"
                        >
                          Bob Ross
                        </span>
                        <span
                          className="date"
                        >
                          4M
                        </span>
                      </div>,
                    ],
                    "className": "authorBox",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "alt": "author of the comment",
                        "src": "pic.jpg",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <span
                            className="name"
                          >
                            Bob Ross
                          </span>,
                          <span
                            className="date"
                          >
                            4M
                          </span>,
                        ],
                        "className": "author",
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "Bob Ross",
                            "className": "name",
                          },
                          "ref": null,
                          "rendered": "Bob Ross",
                          "type": "span",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "4M",
                            "className": "date",
                          },
                          "ref": null,
                          "rendered": "4M",
                          "type": "span",
                        },
                      ],
                      "type": "div",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Architecto autem nam quae delectus maxime inventore odio libero provident obcaecati a.",
                    "className": "content",
                  },
                  "ref": null,
                  "rendered": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Architecto autem nam quae delectus maxime inventore odio libero provident obcaecati a.",
                  "type": "p",
                },
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="authorBox"
                  >
                    <img
                      alt="author of the comment"
                      src="pic.jpg"
                    />
                    <div
                      className="author"
                    >
                      <span
                        className="name"
                      >
                        Mike Ross
                      </span>
                      <span
                        className="date"
                      >
                        13d
                      </span>
                    </div>
                  </div>,
                  <p
                    className="content"
                  >
                    Lorem ipsum dolor sit amet consectetur adipisicing elit. Architecto autem nam quae delectus maxime inventore odio libero provident obcaecati a.
                  </p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <img
                        alt="author of the comment"
                        src="pic.jpg"
                      />,
                      <div
                        className="author"
                      >
                        <span
                          className="name"
                        >
                          Mike Ross
                        </span>
                        <span
                          className="date"
                        >
                          13d
                        </span>
                      </div>,
                    ],
                    "className": "authorBox",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "alt": "author of the comment",
                        "src": "pic.jpg",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <span
                            className="name"
                          >
                            Mike Ross
                          </span>,
                          <span
                            className="date"
                          >
                            13d
                          </span>,
                        ],
                        "className": "author",
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "Mike Ross",
                            "className": "name",
                          },
                          "ref": null,
                          "rendered": "Mike Ross",
                          "type": "span",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "13d",
                            "className": "date",
                          },
                          "ref": null,
                          "rendered": "13d",
                          "type": "span",
                        },
                      ],
                      "type": "div",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Architecto autem nam quae delectus maxime inventore odio libero provident obcaecati a.",
                    "className": "content",
                  },
                  "ref": null,
                  "rendered": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Architecto autem nam quae delectus maxime inventore odio libero provident obcaecati a.",
                  "type": "p",
                },
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="inputComment"
              id="inputComment"
              onKeyPress={[MockFunction]}
              placeholder="Add a comment"
              type="text"
            />,
            "className": "inputBox",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "inputComment",
              "id": "inputComment",
              "onKeyPress": [MockFunction],
              "placeholder": "Add a comment",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
